### ISSUE TITLE
How to deploy split horizon using ACME certificates?

### ISSUE THREAD

**Issue by rrileyca** [mongodb/mongodb-kubernetes-operator#1285]
In the split horizon documents, the kubernetes Service names must be in the SAN of a certificate. Using ACME, this is impossible and no public certificate authority will issue such a certificate. 

Question is: How can you deploy Mongo as a replicaset, using PKI-issued certificates, so that inter-cluster TLS communication and externalAccess (connection from outside kubernetes) will work properly and securely?

My thinking is to get PKI issued certificates issued for `mongodb-0.mydomain.com`, and somehow convince Mongo that the hostname is itself (127.0.0.1 or some other address that it is bound to).

**Comment by jeffreyflynt** [comment]
I was able to do it with this tutorial. I just went through the same issue this week.

https://antonputra.com/kubernetes/how-to-install-mongodb-on-kubernetes/#secure-mongodb-with-tlsssl

**Comment by rrileyca** [comment]
@jeffreyflynt sorry but that article not only references another Helm chart, but it uses the `cluster.local` domain in the Subject Alternative Names in the certificate, which you can't get signed by a public CA (because you cannot satisfy proof of ownership over the domain).

**Comment by jeffreyflynt** [comment]
Ah, I totally missed the ACME part. But in your use case, we did this in a spike at another company. This article should get you going: https://mohaamer5.medium.com/why-and-how-to-connect-mongodb-servers-across-different-k8s-clusters-65c1e5cb505a

We used an internal load balancer and in the article it talks about that like so:

_You can route traffic directly to the node by setting the Service attribute externalTrafficPolicy to Local. This forces the Service to only proxy traffic for local endpoints. Meaning, any node in the cluster that is not running the specific instance of the Pod will not proxy traffic and will fail any external connections._

_I need to add custom entries to /etc/hosts in pods/containers to resolve MongoDB replicaset’s hosts config._

I remember doing it exactly this way to get this setup. Basically changing the local hostname to something not local.
